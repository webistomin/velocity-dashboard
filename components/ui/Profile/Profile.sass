+b('profile')
  +e('container')
    +media($screen-lg)
      max-width: calc(100% - 35.6rem)
      margin: 0

  +e('grid')
    display: grid
    justify-content: start
    grid-template-columns: repeat(12, 1fr)
    grid-gap: rem(14px)

    +media($screen-sm)
      grid-gap: rem(24px)

    +media($screen-lg)
      grid-gap: rem(34px)

  +e('block')
    &:first-of-type
      grid-column: 12 span
      grid-row: 1 span

    &:nth-of-type(2)
      grid-column: 12 span
      grid-row: 1 span

    &:nth-of-type(3)
      grid-column: 12 span
      grid-row: 1 span

    &:nth-of-type(4)
      grid-column: 12 span
      grid-row: 4 span

    +media($screen-sm)
      &:first-of-type
        grid-column: 6 span
        grid-row: 1 span

      &:nth-of-type(2)
        grid-column: 6 span
        grid-row: 1 span

    +media($screen-lg)
      &:first-of-type
        grid-column: 4 span
        grid-row: 1 span

      &:nth-of-type(2)
        grid-column: 4 span
        grid-row: 1 span

      &:nth-of-type(3)
        grid-column: 4 span
        grid-row: 1 span

  +e('scroller')
    &.base-table__scroller
      height: calc(100vh - 380px)
