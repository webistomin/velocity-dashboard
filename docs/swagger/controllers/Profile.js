/* eslint-disable @typescript-eslint/no-unused-vars */
'use strict';

const utils = require('../utils/writer.js');
const Profile = require('../service/ProfileService');

module.exports.profileAvatar = function profileAvatar(req, res, next, body) {
  Profile.profileAvatar(body)
    .then(function (response) {
      utils.writeJson(res, response);
    })
    .catch(function (response) {
      utils.writeJson(res, response);
    });
};

module.exports.profileOwn = function profileOwn(req, res, next, authorization) {
  Profile.profileOwn(authorization)
    .then(function (response) {
      utils.writeJson(res, response);
    })
    .catch(function (response) {
      utils.writeJson(res, response);
    });
};

module.exports.profileUpdate = function profileUpdate(req, res, next, body) {
  Profile.profileUpdate(body)
    .then(function (response) {
      utils.writeJson(res, response);
    })
    .catch(function (response) {
      utils.writeJson(res, response);
    });
};
