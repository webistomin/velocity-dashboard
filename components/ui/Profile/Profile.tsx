import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import BaseBlock from 'components/base/BaseBlock';
import BaseStat from 'components/base/BaseStat';
import BaseTable from 'components/base/BaseTable';

import './Profile.sass';

@Component({
  name: 'Profile',
})
export default class Profile extends VueComponent {
  public tableData = [
    {
      id: '#2178',
      subject: 'Refund request',
      date: '05/04/2018',
      status: 'Active',
    },
    {
      id: '#2179',
      subject: 'Active request',
      date: '05/04/2018',
      status: 'Active',
    },
    {
      id: '#2180',
      subject: 'Refund request',
      date: '05/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2181',
      subject: 'Refund request',
      date: '05/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2182',
      subject: 'Refund request',
      date: '05/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2183',
      subject: 'Refund request',
      date: '05/11/2018',
      status: 'Active',
    },
    {
      id: '#2184',
      subject: 'Active request',
      date: '05/03/2018',
      status: 'Active',
    },
    {
      id: '#2185',
      subject: 'Refund request',
      date: '02/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2186',
      subject: 'Refund request',
      date: '05/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2187',
      subject: 'Refund request',
      date: '01/04/2019',
      status: 'Unresolved',
    },
  ];

  public tableConfig = [
    {
      key: 'id',
      title: 'Id',
    },
    {
      key: 'subject',
      title: 'Subject',
    },
    {
      key: 'date',
      title: 'Latest update',
    },
    {
      key: 'status',
      title: 'Status',
    },
  ];

  public render(): VNode {
    return (
      <section class='profile'>
        <div class='profile__container container'>
          <div class='profile__grid'>
            <BaseBlock title='Resolved tickets' class='profile__block'>
              <BaseStat value={1233} color='purple' icon='icon-mail' align='row' />
            </BaseBlock>
            <BaseBlock title='Average response' class='profile__block'>
              <BaseStat value={12} measure='m' color='darkturquoise' icon='icon-bubble' align='row' />
            </BaseBlock>
            <BaseBlock title='Median NPS' class='profile__block'>
              <BaseStat value={4.32} color='green' icon='icon-calendar' align='row' />
            </BaseBlock>
            <BaseBlock simple={true} class='profile__block'>
              <BaseTable tableConfig={this.tableConfig} tableData={this.tableData} />
            </BaseBlock>
          </div>
        </div>
      </section>
    );
  }
}
