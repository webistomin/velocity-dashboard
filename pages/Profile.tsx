import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import Profile from 'components/ui/Profile';

@Component({
  name: 'ProfilePage',
})
export default class ProfilePage extends VueComponent {
  render(): VNode {
    return (
      <main class='page-content page-content_offset'>
        <Profile />
      </main>
    );
  }
}
