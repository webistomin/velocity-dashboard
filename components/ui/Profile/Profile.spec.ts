import { shallowMount } from '@vue/test-utils';
import Profile from './Profile';

describe('Profile UI', () => {
  const factory = (propsData = {}, options = {}) => {
    return shallowMount(Profile, {
      propsData: {
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
