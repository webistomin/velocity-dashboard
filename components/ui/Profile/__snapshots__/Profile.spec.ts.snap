// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile UI Matches snapshot 1`] = `
"<section class=\\"profile\\">
  <div class=\\"profile__container container\\">
    <div class=\\"profile__grid\\">
      <baseblock-stub title=\\"Resolved tickets\\" class=\\"profile__block\\">
        <basestat-stub value=\\"1233\\" prevvalue=\\"0\\" icon=\\"icon-mail\\" color=\\"purple\\" align=\\"row\\"></basestat-stub>
      </baseblock-stub>
      <baseblock-stub title=\\"Average response\\" class=\\"profile__block\\">
        <basestat-stub value=\\"12\\" prevvalue=\\"0\\" measure=\\"m\\" icon=\\"icon-bubble\\" color=\\"darkturquoise\\" align=\\"row\\"></basestat-stub>
      </baseblock-stub>
      <baseblock-stub title=\\"Median NPS\\" class=\\"profile__block\\">
        <basestat-stub value=\\"4.32\\" prevvalue=\\"0\\" icon=\\"icon-calendar\\" color=\\"green\\" align=\\"row\\"></basestat-stub>
      </baseblock-stub>
      <baseblock-stub issimple=\\"true\\" class=\\"profile__block\\">
        <basetable-stub tableconfig=\\"[object Object],[object Object],[object Object],[object Object]\\" tabledata=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" scrollermix=\\"profile__scroller\\"></basetable-stub>
      </baseblock-stub>
    </div>
  </div>
</section>"
`;
